src/www/ui/api/Controllers/UploadTreeController.php

```diff
@@ -229,4 +265,149 @@ public function getBulkHistory($request, $response, $args)
     }
     return $response->withJson($updatedRes, 200);
   }
+
+  /**
+   * Get all license decisions for a particular upload-tree
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function getLicenseDecisions($request, $response, $args)
+  {
+    $uploadTreeId = intval($args['itemId']);
+    $uploadPk = intval($args['id']);
+    $returnVal = null;
+    $uploadDao = $this->restHelper->getUploadDao();
+    $licenses = [];
+
+    try {
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadPk)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if (!$this->dbHelper->doesIdExist($uploadDao->getUploadtreeTableName($uploadPk), "uploadtree_pk", $uploadTreeId)) {
+        $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+      }
+
+      if ($returnVal !== null) {
+        return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+      }
+
+      $itemTreeBounds = $uploadDao->getItemTreeBoundsFromUploadId($uploadTreeId, $uploadPk);
```

Comment:
Check if `$itemTreeBounds->containsFiles()` is true. If yes, then this endpoint should return `400` as it is expected to work only for files and not for folders.

The message can be "Item expected to be a file, container sent."